### PR TITLE
make CMake's version config file architecture-independent

### PR DESCRIPTION
**Pull request by uhoreg**
Fixes https://github.com/nlohmann/json/issues/1697. In short, CMake's `write_basic_package_version_file` by default creates a file that checks whether it is being called on a 32- or 64-bit architecture, and will fail if it is called on a different bit size than what it was built on, even though that doesn't affect the JSON library, since it is header-only.  This replaces the automatically-generated file with a file that checks the version requirement, but does not check the word size.  (It is essentially CMake's file, but with the word size check removed.)

### CODE REVIEW

**Comment by t-b** [comment]
@uhoreg @nlohmann  Is that good to go? I *think* I'm seeing some fallout of not having that in #1780.

**Comment by uhoreg** [comment]
@nlohmann can you re-open this please (or even better, merge it. ;) )

**Comment by uhoreg** [comment]
not stale

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,8 +89,10 @@ endif()
 ## install header files, generate and install cmake config files for find_package()
 ##
 include(CMakePackageConfigHelpers)
-write_basic_package_version_file(
-    ${NLOHMANN_JSON_CMAKE_VERSION_CONFIG_FILE} COMPATIBILITY SameMajorVersion
+configure_file(
```
Comment (nlohmann) [inline CMakeLists.txt]: Could you add a comment referencing issue #1697?

File: cmake/nlohmann_jsonConfigVersion.cmake.in [hunk 2]
```diff
@@ -0,0 +1,16 @@
+set(PACKAGE_VERSION "@PROJECT_VERSION@")
```
Comment (nlohmann) [inline cmake/nlohmann_jsonConfigVersion.cmake.in]: Could you please explain in a comment why this file is needed (as I understood, to simulate CMake 3.14 behavior)?

**Comment by uhoreg** [comment]
I've added comments as requested.

**Comment by nlohmann** [comment]
Thanks!

**Comment by nlohmann** [comment]
* * *

### 🔖 Release item

This issue/PR will be part of the next release of the library. This template helps preparing the release notes.

#### Type

- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] ⚡️ Improvement
- [ ] 🔨 Further Change
- [ ] 🔥 Deprecated function

#### Description

- Made CMake's version config file architecture-independent. #1697 #1746

* * *

**Pull request merged**